dask_ml/cluster/k_means.py

```diff
@@ -156,10 +156,8 @@ def _check_array(self, X):
         if isinstance(X, pd.DataFrame):
             X = X.values
 
-        elif isinstance(X, dd.DataFrame):
-            raise TypeError(
-                "Cannot fit on dask.dataframe due to unknown " "partition lengths."
-            )
+        if isinstance(X, dd.DataFrame):
+            X = X.to_dask_array(True)
```

Comment:
style-nit: specify the `lengths=` keyword for boolean arguments.

LGTM otherwise.